server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2156,6 +2157,11 @@ public void waitForRemoteStoreSync(Runnable onProgress) {
                 return;
             }
         }
+        throw new IOException(
+            "Failed to upload to remote segment store within remote upload timeout of "
+                + getRecoverySettings().internalRemoteUploadTimeout().getMinutes()
```

Comment:
Can we write a corresponding test that simulates a timeout?

Reply:
There is already an IT `testCreateCloneIndexFailure` which simulates the timeout . I have added more asserts to same.